src/NuGet.Core/NuGet.Packaging/Signing/Timestamp/Rfc3161TimestampProvider.cs

```diff
@@ -129,11 +135,19 @@ public byte[] TimestampData(TimestampRequest request, ILogger logger, Cancellati
 
                     if (!timestampCertChain.Build(timestampSignerCertificate))
                     {
+                        var messages = timestampCertChain.ChainStatus
```

Comment:
I would move this to a shared helper method for converting the chain status into messages since it is used above also.